### PR TITLE
tests: drivers: watchdog: Run tests on nRF54L20pdk

### PR DESCRIPTION
**Pull request by nordic-segl**
Add overlays and align watchdog tests to be executed on nrf54L20pdk.

### CODE REVIEW

File: tests/drivers/watchdog/wdt_error_cases/src/main.c [hunk 1]
```diff
@@ -43,7 +43,8 @@
 
 /* Align tests to the specific target: */
 #if defined(CONFIG_SOC_NRF54L05) || defined(CONFIG_SOC_NRF54L10) || \
-	defined(CONFIG_SOC_NRF54L15) || defined(CONFIG_SOC_NRF54H20) || defined(CONFIG_SOC_NRF9280)
+	defined(CONFIG_SOC_NRF54L15) || defined(CONFIG_SOC_NRF54L20) || \
```
Comment (masz-nordic) [inline tests/drivers/watchdog/wdt_error_cases/src/main.c:46]: `CONFIG_SOC_SERIES_NRF54LX`

Comment (nordic-segl) [reply tests/drivers/watchdog/wdt_error_cases/src/main.c]: Fixed, thanks!

**Review by masz-nordic — DISMISSED** [review]

File: tests/drivers/watchdog/wdt_error_cases/src/main.c [hunk 2]
```diff
@@ -42,7 +42,8 @@
 #define DEFAULT_WINDOW_MIN (0U)
 
 /* Align tests to the specific target: */
-#if defined(CONFIG_SOC_SERIES_NRF54LX) || defined(CONFIG_SOC_NRF9280)
+#if defined(CONFIG_SOC_SERIES_NRF54LX) || defined(CONFIG_SOC_NRF54H20) || \
```
Comment (e-rk) [inline tests/drivers/watchdog/wdt_error_cases/src/main.c:45]: The commit is about `nrf54l20` but `nrf54h20` is being added. Is this intentional?

Comment (nordic-segl) [reply tests/drivers/watchdog/wdt_error_cases/src/main.c:45]: nRF54H20 was removed by mistake in this commit:
https://github.com/zephyrproject-rtos/zephyr/commit/07ee796c7d40fe8a7ccc8198aa3ef53c2f371dae#diff-0301c1f4e2a8750e8bd567b1fd052d0d795bfac98cfdfbdbc98504aa81bb8b51L46

Comment (nordic-segl) [reply tests/drivers/watchdog/wdt_error_cases/src/main.c:45]: I can put this to a separate commit if requested.

**Review by e-rk — APPROVED** [review]

**Review by nordic-piks — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Review by kl-cruz — APPROVED** [review]

**Pull request merged**